<!DOCTYPE html><html lang="zn-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><title>重测序分析脚本 | 🧬小蓝哥的知识荒原🧬</title><meta name="keywords" content="Linux,生物信息学"><meta name="author" content="小蓝哥"><meta name="copyright" content="小蓝哥"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta name="description" content="重测序与群体遗传学 云南农业大学  云南生物资源保护与利用国家重点实验室 （李详  2019年12月31日）  软件安装 该部分软件很难用conda直接安装，安装步骤比较特殊。">
<meta property="og:type" content="article">
<meta property="og:title" content="重测序分析脚本">
<meta property="og:url" content="http://example.com/2021/11/23/%E9%87%8D%E6%B5%8B%E5%BA%8F%E5%88%86%E6%9E%90%E8%84%9A%E6%9C%AC/index.html">
<meta property="og:site_name" content="🧬小蓝哥的知识荒原🧬">
<meta property="og:description" content="重测序与群体遗传学 云南农业大学  云南生物资源保护与利用国家重点实验室 （李详  2019年12月31日）  软件安装 该部分软件很难用conda直接安装，安装步骤比较特殊。">
<meta property="og:locale" content="zn_CN">
<meta property="og:image" content="http://example.com/img/cover/popgene.jpg">
<meta property="article:published_time" content="2021-11-23T06:05:26.000Z">
<meta property="article:modified_time" content="2021-11-23T08:46:53.877Z">
<meta property="article:author" content="小蓝哥">
<meta property="article:tag" content="Linux">
<meta property="article:tag" content="生物信息学">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="http://example.com/img/cover/popgene.jpg"><link rel="shortcut icon" href="/img/favicon.png"><link rel="canonical" href="http://example.com/2021/11/23/%E9%87%8D%E6%B5%8B%E5%BA%8F%E5%88%86%E6%9E%90%E8%84%9A%E6%9C%AC/"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css" media="print" onload="this.media='all'"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/node-snackbar/dist/snackbar.min.css" media="print" onload="this.media='all'"><script>const GLOBAL_CONFIG = { 
  root: '/',
  algolia: undefined,
  localSearch: {"path":"search.xml","languages":{"hits_empty":"We didn't find any results for the search: ${query}"}},
  translate: {"defaultEncoding":1,"translateDelay":0,"msgToTraditionalChinese":"繁","msgToSimplifiedChinese":"简"},
  noticeOutdate: undefined,
  highlight: {"plugin":"highlighjs","highlightCopy":true,"highlightLang":true,"highlightHeightLimit":200},
  copy: {
    success: 'Copy successfully',
    error: 'Copy error',
    noSupport: 'The browser does not support'
  },
  relativeDate: {
    homepage: true,
    post: true
  },
  runtime: 'days',
  date_suffix: {
    just: 'Just',
    min: 'minutes ago',
    hour: 'hours ago',
    day: 'days ago',
    month: 'months ago'
  },
  copyright: {"limitCount":50,"languages":{"author":"Author: 小蓝哥","link":"Link: ","source":"Source: 🧬小蓝哥的知识荒原🧬","info":"Copyright is owned by the author. For commercial reprints, please contact the author for authorization. For non-commercial reprints, please indicate the source."}},
  lightbox: 'fancybox',
  Snackbar: {"chs_to_cht":"Traditional Chinese Activated Manually","cht_to_chs":"Simplified Chinese Activated Manually","day_to_night":"Dark Mode Activated Manually","night_to_day":"Light Mode Activated Manually","bgLight":"#49b1f5","bgDark":"#2d3035","position":"bottom-left"},
  source: {
    jQuery: 'https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js',
    justifiedGallery: {
      js: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/js/jquery.justifiedGallery.min.js',
      css: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/css/justifiedGallery.min.css'
    },
    fancybox: {
      js: 'https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js',
      css: 'https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.css'
    }
  },
  isPhotoFigcaption: false,
  islazyload: false,
  isanchor: false
}</script><script id="config-diff">var GLOBAL_CONFIG_SITE = {
  title: '重测序分析脚本',
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isToc: true,
  postUpdate: '2021-11-23 16:46:53'
}</script><noscript><style type="text/css">
  #nav {
    opacity: 1
  }
  .justified-gallery img {
    opacity: 1
  }

  #recent-posts time,
  #post-meta time {
    display: inline !important
  }
</style></noscript><script>(win=>{
    win.saveToLocal = {
      set: function setWithExpiry(key, value, ttl) {
        if (ttl === 0) return
        const now = new Date()
        const expiryDay = ttl * 86400000
        const item = {
          value: value,
          expiry: now.getTime() + expiryDay,
        }
        localStorage.setItem(key, JSON.stringify(item))
      },

      get: function getWithExpiry(key) {
        const itemStr = localStorage.getItem(key)

        if (!itemStr) {
          return undefined
        }
        const item = JSON.parse(itemStr)
        const now = new Date()

        if (now.getTime() > item.expiry) {
          localStorage.removeItem(key)
          return undefined
        }
        return item.value
      }
    }
  
    win.getScript = url => new Promise((resolve, reject) => {
      const script = document.createElement('script')
      script.src = url
      script.async = true
      script.onerror = reject
      script.onload = script.onreadystatechange = function() {
        const loadState = this.readyState
        if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
        script.onload = script.onreadystatechange = null
        resolve()
      }
      document.head.appendChild(script)
    })
  
      win.activateDarkMode = function () {
        document.documentElement.setAttribute('data-theme', 'dark')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#0d0d0d')
        }
      }
      win.activateLightMode = function () {
        document.documentElement.setAttribute('data-theme', 'light')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#ffffff')
        }
      }
      const t = saveToLocal.get('theme')
    
          if (t === 'dark') activateDarkMode()
          else if (t === 'light') activateLightMode()
        
      const asideStatus = saveToLocal.get('aside-status')
      if (asideStatus !== undefined) {
        if (asideStatus === 'hide') {
          document.documentElement.classList.add('hide-aside')
        } else {
          document.documentElement.classList.remove('hide-aside')
        }
      }
    
    const fontSizeVal = saveToLocal.get('global-font-size')
    if (fontSizeVal !== undefined) {
      document.documentElement.style.setProperty('--global-font-size', fontSizeVal + 'px')
    }
    
    const detectApple = () => {
      if (GLOBAL_CONFIG_SITE.isHome && /iPad|iPhone|iPod|Macintosh/.test(navigator.userAgent)){
        document.documentElement.classList.add('apple')
      }
    }
    detectApple()
    })(window)</script><meta name="generator" content="Hexo 5.4.0"></head><body><div id="loading-box"><div class="loading-left-bg"></div><div class="loading-right-bg"></div><div class="spinner-box"><div class="configure-border-1"><div class="configure-core"></div></div><div class="configure-border-2"><div class="configure-core"></div></div><div class="loading-word">Loading...</div></div></div><div id="web_bg"></div><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="avatar-img is-center"><img src="/img/head.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="site-data"><div class="data-item is-center"><div class="data-item-link"><a href="/archives/"><div class="headline">Articles</div><div class="length-num">74</div></a></div></div><div class="data-item is-center"><div class="data-item-link"><a href="/tags/"><div class="headline">Tags</div><div class="length-num">11</div></a></div></div><div class="data-item is-center"><div class="data-item-link"><a href="/categories/"><div class="headline">Categories</div><div class="length-num">11</div></a></div></div></div><hr/><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fa fa-home"></i><span> 首页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fa fa-archive"></i><span> 时间轴</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fa fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fa fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fa fa-link"></i><span> 友链</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fa fa-heart"></i><span> 关于</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg" id="page-header" style="background-image: url('/img/cover/popgene.jpg')"><nav id="nav"><span id="blog_name"><a id="site-name" href="/">🧬小蓝哥的知识荒原🧬</a></span><div id="menus"><div id="search-button"><a class="site-page social-icon search"><i class="fas fa-search fa-fw"></i><span> Search</span></a></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fa fa-home"></i><span> 首页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fa fa-archive"></i><span> 时间轴</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fa fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fa fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fa fa-link"></i><span> 友链</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fa fa-heart"></i><span> 关于</span></a></div></div><div id="toggle-menu"><a class="site-page"><i class="fas fa-bars fa-fw"></i></a></div></div></nav><div id="post-info"><h1 class="post-title">重测序分析脚本</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">Created</span><time class="post-meta-date-created" datetime="2021-11-23T06:05:26.000Z" title="Created 2021-11-23 14:05:26">2021-11-23</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">Updated</span><time class="post-meta-date-updated" datetime="2021-11-23T08:46:53.877Z" title="Updated 2021-11-23 16:46:53">2021-11-23</time></span><span class="post-meta-categories"><span class="post-meta-separator">|</span><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6-%E7%BE%A4%E4%BD%93%E9%81%97%E4%BC%A0%E5%AD%A6-Linux/">生物信息学 - 群体遗传学 - Linux</a></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span class="post-meta-wordcount"><i class="far fa-file-word fa-fw post-meta-icon"></i><span class="post-meta-label">Word count:</span><span class="word-count">3.2k</span><span class="post-meta-separator">|</span><i class="far fa-clock fa-fw post-meta-icon"></i><span class="post-meta-label">Reading time:</span><span>18min</span></span><span class="post-meta-separator">|</span><span class="post-meta-pv-cv" id="" data-flag-title="重测序分析脚本"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">Post View:</span><span id="busuanzi_value_page_pv"></span></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="post-content" id="article-container"><center><font face=arial bold size=10 color=black>重测序与群体遗传学</font></center>
<center><font size=4 color=red>云南农业大学  云南生物资源保护与利用国家重点实验室</font></center>
<center><font size=4 color=red>（李详  2019年12月31日）</font></center>
<hr>
<h1>软件安装</h1>
<p>该部分软件很难用<code>conda</code>直接安装，安装步骤比较特殊。</p>
<span id="more"></span>
<h2 id="lumpy-sv"><a class="header-anchor" href="#lumpy-sv">¶</a>lumpy-sv</h2>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">lumpy-sv 依赖 Python2.7，切换到py27环境下进行安装。</span><br><span class="line">conda activate py27</span><br><span class="line">git clone --recursive https://github.com/arq5x/lumpy-sv.git</span><br><span class="line">cd lumpy-sv</span><br><span class="line">make</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 安装svtyper</span></span></span><br><span class="line">conda install svtyper</span><br></pre></td></tr></table></figure>
<h2 id="cnvnator"><a class="header-anchor" href="#cnvnator">¶</a>cnvnator</h2>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">cnvnator依赖于root软件包及samtools软件包(包含HTSlib)，因此先安装Samtools及root软件包。</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 安装samtools</span></span></span><br><span class="line">wget https://github.com/samtools/samtools/releases/download/1.9/samtools-1.9.tar.bz2</span><br><span class="line">tar -xvf samtools-1.9.tar.bz2</span><br><span class="line">cd samtools-1.9/</span><br><span class="line">./configure</span><br><span class="line">make</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 安装root软件, 直接下载后解压即可</span></span></span><br><span class="line">wget https://root.cern/download/root_v6.18.04.Linux-ubuntu18-x86_64-gcc7.4.tar.gz</span><br><span class="line">tar -zxvf root_v6.18.04.Linux-ubuntu18-x86_64-gcc7.4.tar.gz</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 安装cnvnator</span></span></span><br><span class="line">git clone https://github.com/abyzovlab/CNVnator.git</span><br><span class="line">cd CNVnator</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 链接samtools软件目录到cnvnator目录</span></span></span><br><span class="line">ln -s ../samtools-1.9 ./samtools</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 加载root软件到环境变量</span></span></span><br><span class="line">export ROOTSYS=/pub/software/root</span><br><span class="line">export PATH=/pub/software/root/bin:$PATH</span><br><span class="line">export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$ROOTSYS/lib</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 安装cnvnator</span></span></span><br><span class="line">make LIBS=&quot;-lcrypto&quot; </span><br><span class="line"></span><br><span class="line">less ../data/sample.list | awk &#x27;&#123; print &quot;bwa mem -t 2 -R &#x27;\&#x27;&#x27;@RG\\tID:&quot;$1&quot;\\tSM:&quot;$1&quot;\\tPL:illumina&#x27;\&#x27;&#x27; ../01.ref/genome.fasta  ../data/&quot;$1&quot;_1.fq.gz ../data/&quot;$1&quot;_2.fq.gz |  /pub/software/samtools/samtools sort -@ 2 -m 1G -o &quot;$1&quot;.sort.bam -&quot; &#125;&#x27;</span><br></pre></td></tr></table></figure>
<h1>变异检测</h1>
<h2 id="构建基因组index"><a class="header-anchor" href="#构建基因组index">¶</a>构建基因组index</h2>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> create link</span></span><br><span class="line">ln -s  ../data/genome.fasta  ./genome.fasta</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> <span class="keyword">for</span> gatk</span></span><br><span class="line">samtools faidx genome.fasta </span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> <span class="keyword">for</span> bwa</span></span><br><span class="line">bwa index  genome.fasta  </span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> <span class="keyword">for</span> picard</span></span><br><span class="line">picard CreateSequenceDictionary R=genome.fasta</span><br></pre></td></tr></table></figure>
<h2 id="比对排序去重"><a class="header-anchor" href="#比对排序去重">¶</a>比对排序去重</h2>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 比对排序</span></span><br><span class="line">bwa mem -t 2 -R &#x27;@RG\tID:S1\tSM:S1\tPL:illumina&#x27; ../01.ref/genome.fasta  ../data/S1_1.fq.gz ../data/S1_2.fq.gz |  /pub/software/samtools/samtools sort -@ 2 -m 1G -o S1.sort.bam -</span><br><span class="line">bwa mem -t 2 -R &#x27;@RG\tID:S2\tSM:S2\tPL:illumina&#x27; ../01.ref/genome.fasta  ../data/S2_1.fq.gz ../data/S2_2.fq.gz |  /pub/software/samtools/samtools sort -@ 2 -m 1G -o S2.sort.bam -</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 去除重复</span></span><br><span class="line">picard -Xmx4g  MarkDuplicates I=S1.sort.bam O=S1.sort.rmdup.bam  CREATE_INDEX=true  REMOVE_DUPLICATES=true M=S1.marked_dup_metrics.txt</span><br><span class="line">picard -Xmx4g  MarkDuplicates I=S2.sort.bam O=S2.sort.rmdup.bam  CREATE_INDEX=true  REMOVE_DUPLICATES=true M=S2.marked_dup_metrics.txt</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 比对率统计</span></span><br><span class="line">/pub/software/samtools/samtools  flagstat  S1.sort.bam &gt; S1.sort.bam.flagstat</span><br><span class="line">/pub/software/samtools/samtools  flagstat  S2.sort.bam &gt; S2.sort.bam.flagstat</span><br></pre></td></tr></table></figure>
<h2 id="SNP、Indel检测"><a class="header-anchor" href="#SNP、Indel检测">¶</a>SNP、Indel检测</h2>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> step1.HaplotypeCaller</span></span><br><span class="line">gatk --java-options &quot;-Xmx10g -Djava.io.tmpdir=./tmp&quot; HaplotypeCaller -R ../01.ref/genome.fasta -I ../02.mapping/S1.sort.rmdup.bam  -ERC GVCF -O S1.g.vcf 1&gt;S1.HC.log   2&gt;&amp;1</span><br><span class="line">gatk --java-options &quot;-Xmx10g -Djava.io.tmpdir=./tmp&quot; HaplotypeCaller -R ../01.ref/genome.fasta -I ../02.mapping/S2.sort.rmdup.bam  -ERC GVCF -O S2.g.vcf 1&gt;S2.HC.log   2&gt;&amp;1</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> step2.CombineGVCFs</span></span><br><span class="line">ls ./S*.g.vcf &gt; gvcf.list</span><br><span class="line">gatk  --java-options &quot;-Xmx4g -Djava.io.tmpdir=./tmp&quot;    CombineGVCFs -R ../01.ref/genome.fasta -V gvcf.list  -O all.merge.g.vcf </span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> step3.GenotypeGVCFs</span></span><br><span class="line">gatk  --java-options &quot;-Xmx4g -Djava.io.tmpdir=./tmp&quot;   GenotypeGVCFs -R ../01.ref/genome.fasta --variant all.merge.g.vcf -O all.merge_raw.vcf</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> step4.SNP</span></span><br><span class="line">gatk  --java-options &quot;-Xmx4g -Djava.io.tmpdir=./tmp&quot;  SelectVariants  -R ../01.ref/genome.fasta -V all.merge_raw.vcf --select-type SNP -O all.raw.snp.vcf</span><br><span class="line"></span><br><span class="line">gatk  --java-options &quot;-Xmx4g -Djava.io.tmpdir=./tmp&quot;  VariantFiltration -R ../01.ref/genome.fasta -V all.raw.snp.vcf --filter-expression &quot;QD &lt; 2.0 || MQ &lt; 40.0 || FS &gt; 60.0 || SOR &gt; 3.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&quot; --filter-name &#x27;SNP_filter&#x27; -O all.filter.snp.vcf</span><br><span class="line"></span><br><span class="line">gatk  --java-options &quot;-Xmx4g -Djava.io.tmpdir=./tmp&quot;  SelectVariants  -R ../01.ref/genome.fasta -V all.filter.snp.vcf --exclude-filtered  -O all.filtered.snp.vcf</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> step5.InDel</span></span><br><span class="line">gatk  --java-options &quot;-Xmx4g -Djava.io.tmpdir=./tmp&quot;  SelectVariants  -R ../01.ref/genome.fasta -V all.merge_raw.vcf --select-type INDEL -O all.raw.indel.vcf</span><br><span class="line"></span><br><span class="line">gatk  --java-options &quot;-Xmx4g -Djava.io.tmpdir=./tmp&quot;  VariantFiltration -R ../01.ref/genome.fasta -V all.raw.indel.vcf --filter-expression &quot;QD &lt; 2.0 || FS &gt; 200.0 || SOR &gt; 10.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&quot; --filter-name &#x27;INDEL_filter&#x27; -O all.filter.indel.vcf</span><br><span class="line"></span><br><span class="line">gatk  --java-options &quot;-Xmx4g -Djava.io.tmpdir=./tmp&quot;  SelectVariants  -R ../01.ref/genome.fasta -V all.filter.indel.vcf  --exclude-filtered   -O all.filtered.indel.vcf</span><br></pre></td></tr></table></figure>
<h2 id="SV检测"><a class="header-anchor" href="#SV检测">¶</a>SV检测</h2>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> step1.discordants_bam</span></span><br><span class="line">/pub/software/samtools/samtools view -b -F 1294 -@ 12 ../02.mapping/S1.sort.rmdup.bam &gt; S1.discordants.bam</span><br><span class="line">/pub/software/samtools/samtools view -b -F 1294 -@ 12 ../02.mapping/S2.sort.rmdup.bam &gt; S2.discordants.bam</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> step2.splitters_bam</span></span><br><span class="line">/pub/software/samtools/samtools view -h ../02.mapping/S1.sort.rmdup.bam | /pub/software/lumpy-sv/scripts/extractSplitReads_BwaMem -i stdin | /pub/software/samtools/samtools  sort -   &gt; S1.splitters.bam</span><br><span class="line">/pub/software/samtools/samtools view -h ../02.mapping/S2.sort.rmdup.bam | /pub/software/lumpy-sv/scripts/extractSplitReads_BwaMem -i stdin | /pub/software/samtools/samtools  sort -   &gt; S2.splitters.bam</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> step3.lumpy</span></span><br><span class="line">/pub/software/lumpy-sv/bin/lumpyexpress -B ../02.mapping/S1.sort.rmdup.bam,../02.mapping/S2.sort.rmdup.bam  -S S1.splitters.bam,S2.splitters.bam   -D S1.discordants.bam,S2.discordants.bam -o all.sv.lumpy.vcf</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> step4.ind_genotype</span></span><br><span class="line">vcftools --vcf all.sv.lumpy.vcf --indv S1 --recode --recode-INFO-all --out S1 &amp;&amp; svtyper -i S1.recode.vcf -B ../02.mapping/S1.sort.rmdup.bam  -o S1.genotype.vcf</span><br><span class="line">vcftools --vcf all.sv.lumpy.vcf --indv S2 --recode --recode-INFO-all --out S2 &amp;&amp; svtyper -i S2.recode.vcf -B ../02.mapping/S2.sort.rmdup.bam  -o S2.genotype.vcf</span><br></pre></td></tr></table></figure>
<h2 id="CNV检测"><a class="header-anchor" href="#CNV检测">¶</a>CNV检测</h2>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> step1.prepare_genome</span></span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># split genome into single sequence</span></span></span><br><span class="line"></span><br><span class="line"> ln -s  ../data/genome.fasta ./genome.fa</span><br><span class="line"></span><br><span class="line"> seqkit split -i ./genome.fa  -O split</span><br><span class="line"></span><br><span class="line"> ls ./split/genome.id_*.fa |sed &#x27;s/\.\/split\/genome\.id_//&#x27; |awk &#x27;&#123;print &quot;mv ./split/genome.id_&quot;$aa&quot;  ./split/&quot;$aa&#125;&#x27; &gt; mv_name.sh</span><br><span class="line"></span><br><span class="line"> sh mv_name.sh</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> step2.cnvnator</span></span><br><span class="line">export ROOTSYS=/pub/software/root</span><br><span class="line">export PATH=/pub/software/root/bin:$PATH</span><br><span class="line">export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$ROOTSYS/lib</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 链接文件到当前目录</span></span></span><br><span class="line">ln -s ../02.mapping/S1.sort.rmdup.bam </span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 提取比对结果</span></span></span><br><span class="line">/pub/software/CNVnator/cnvnator -genome genome.fa -root S1.root -tree S1.sort.rmdup.bam</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#生成深度分布</span></span></span><br><span class="line">/pub/software/CNVnator/cnvnator -genome genome.fa -root S1.root -his 500  -d split</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 进行统计计算</span></span></span><br><span class="line">/pub/software/CNVnator/cnvnator -root S1.root -stat 500</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 检查bin size是否合适</span></span></span><br><span class="line">/pub/software/CNVnator/cnvnator -root S1.root -eval 500  &gt; S1.eval.ratio </span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># RD信号分割</span></span></span><br><span class="line">/pub/software/CNVnator/cnvnator -root S1.root -partition 500</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 进行CNV检测</span></span></span><br><span class="line">/pub/software/CNVnator/cnvnator -root S1.root  -call 500 &gt; S1.cnv</span><br></pre></td></tr></table></figure>
<h2 id="SNP-INDEL注释"><a class="header-anchor" href="#SNP-INDEL注释">¶</a>SNP INDEL注释</h2>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> step1.prepare_genome</span></span><br><span class="line">ln -s ../data/genome.gtf </span><br><span class="line">ln -s ../data/genome.fasta</span><br><span class="line"></span><br><span class="line">gtfToGenePred  -genePredExt genome.gtf genome_refGene.txt</span><br><span class="line">/pub/software/annovar/retrieve_seq_from_fasta.pl --format refGene --seqfile genome.fasta genome_refGene.txt --out genome_refGeneMrna.fa</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash">step2.ann_vcf</span></span><br><span class="line">ln -s  ../03.SNP_indel/all.filtered.snp.vcf </span><br><span class="line"></span><br><span class="line">/pub/software/annovar/convert2annovar.pl -format vcf4old all.filtered.snp.vcf  &gt;all.snp.vcf.annovar.input</span><br><span class="line"></span><br><span class="line">/pub/software/annovar/annotate_variation.pl -geneanno --neargene 2000 -buildver genome -dbtype refGene -outfile all.anno -exonsort all.snp.vcf.annovar.input ./</span><br></pre></td></tr></table></figure>
<h2 id="CNV和SV注释"><a class="header-anchor" href="#CNV和SV注释">¶</a>CNV和SV注释</h2>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">less -S ../04.SV/all.sv.lumpy.vcf|awk -F &quot;;&quot; &#x27;&#123; if($1!~/#/)&#123;print $1&quot;\t&quot;$2&quot;\t&quot;$3&quot;\t&quot;$4&#125; &#125;&#x27;|awk &#x27;&#123;print $1&quot;\t&quot;$2&quot;\t&quot;$11&quot;\t&quot;$3&quot;\t&quot;$8&#125;&#x27;|sed &#x27;s/END=//&#x27; |sed &#x27;s/SVTYPE=//&#x27;| awk &#x27;$3&lt;$2&#x27; &gt; SV.bed</span><br><span class="line">less -S ../04.SV/all.sv.lumpy.vcf|awk -F &quot;;&quot; &#x27;&#123; if($1!~/#/)&#123;print $1&quot;\t&quot;$2&quot;\t&quot;$3&quot;\t&quot;$4&#125; &#125;&#x27;|awk &#x27;&#123;print $1&quot;\t&quot;$2&quot;\t&quot;$11&quot;\t&quot;$3&quot;\t&quot;$8&#125;&#x27;|sed &#x27;s/END=//&#x27; |sed &#x27;s/SVTYPE=//&#x27;  &gt; SV.bed</span><br><span class="line">awk &#x27;$5 !~ /BND/&#x27; SV.bed &gt;  SV_filter.bed</span><br><span class="line">bedtools  intersect -wo  -a SV_filter.bed -b gene.gtf  &gt; SV_filter.Anno</span><br><span class="line">less -S ../05.CNV/S1.cnv | awk &#x27;&#123;print $2 &quot;\t&quot; $1&#125;&#x27;|sed &#x27;s/:/\t/&#x27;|sed &#x27;s/-/\t/&#x27; &gt; CNV.bed</span><br><span class="line">bedtools  intersect -wo  -a  CNV.bed  -b gene.gtf &gt; CNV.Anno</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1>群体结构</h1>
<h2 id="过滤"><a class="header-anchor" href="#过滤">¶</a>过滤</h2>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">vcf=../data/all.vcf</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># filter missing and maf</span></span></span><br><span class="line">plink --vcf  $vcf --geno 0.1 --maf 0.01 --out all.missing_maf --recode vcf-iid  --allow-extra-chr --set-missing-var-ids @:# --keep-allele-order</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># filter LD</span></span> </span><br><span class="line">plink --vcf  all.missing_maf.vcf  --indep-pairwise 50 10 0.2 --out tmp.ld   --allow-extra-chr --set-missing-var-ids @:# </span><br><span class="line">plink --vcf  all.missing_maf.vcf  --make-bed --extract tmp.ld.prune.in  --out all.LDfilter --recode vcf-iid  --keep-allele-order  --allow-extra-chr --set-missing-var-ids @:#  </span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h2 id="建树"><a class="header-anchor" href="#建树">¶</a>建树</h2>
<h3 id="NJ"><a class="header-anchor" href="#NJ">¶</a>NJ</h3>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">perl  ../../script/vcf2phy.pl  ../../00.filter/all.LDfilter.vcf  &gt; sequences.dat</span><br><span class="line"></span><br><span class="line">echo -e &quot;sequences.dat\nY&quot; &gt; dnadist.cfg</span><br><span class="line"></span><br><span class="line">dnadist &lt; dnadist.cfg  &gt;dnadist.log</span><br><span class="line"></span><br><span class="line">mv outfile infile.dist</span><br><span class="line"></span><br><span class="line">echo -e &quot;infile.dist\nY&quot;  &gt; neighbor.cfg</span><br><span class="line"></span><br><span class="line">neighbor  &lt;  neighbor.cfg  &gt;nj.log</span><br><span class="line"></span><br><span class="line">less infile.dist | tr &#x27;\n&#x27; &#x27;|&#x27;| sed &#x27;s/| / /g&#x27; | tr &#x27;|&#x27; &#x27;\n&#x27; &gt;infile.dist.table</span><br><span class="line">less outtree | tr &#x27;\n&#x27; &#x27; &#x27;|sed &#x27;s/ //g&#x27; &gt; outtree.nwk</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h3 id="tree-snphylo"><a class="header-anchor" href="#tree-snphylo">¶</a>tree_snphylo</h3>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">/pub/software/SNPhylo/snphylo.sh -v ../../00.filter/all.LDfilter.vcf -r -l 1 -m 0.01 -o GA0001 -b -B 2 -P tree_snphylo </span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h2 id="PCA"><a class="header-anchor" href="#PCA">¶</a>PCA</h2>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">plink --vcf  ../00.filter/all.LDfilter.vcf --pca 10 --out  PCA_out   --allow-extra-chr --set-missing-var-ids @:# </span><br><span class="line"></span><br><span class="line">Rscript ../script/draw_PCA.R  PCA_out.eigenvec 1 2  ../data/sample.pop  PCA_out_figure</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h2 id="群体结构"><a class="header-anchor" href="#群体结构">¶</a>群体结构</h2>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> step0.vcf2bed</span></span><br><span class="line">plink --vcf  ../00.filter/all.LDfilter.vcf  --make-bed --out all  --allow-extra-chr --keep-allele-order --set-missing-var-ids @:#</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> step1.get_Admix_shell</span></span><br><span class="line">seq 2 4 | awk &#x27;&#123;print &quot;admixture --cv -j2 all.bed &quot;$1&quot; 1&gt;admix.&quot;$1&quot;.log 2&gt;&amp;1&quot;&#125;&#x27; &gt; admixture.sh</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> step2.run_admix</span></span><br><span class="line">sh admixture.sh</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> step3.draw_structure</span></span><br><span class="line">mkdir result</span><br><span class="line">cp  ./*.Q result/</span><br><span class="line">Rscript  ../script/draw_admixture.R result all.nosex  structure </span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> admixture</span></span><br><span class="line">admixture --cv -j2 all.bed 2 1&gt;admix.2.log 2&gt;&amp;1</span><br><span class="line">admixture --cv -j2 all.bed 3 1&gt;admix.3.log 2&gt;&amp;1</span><br><span class="line">admixture --cv -j2 all.bed 4 1&gt;admix.4.log 2&gt;&amp;1</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h2 id="遗传衰减"><a class="header-anchor" href="#遗传衰减">¶</a>遗传衰减</h2>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> step1.LD_calcu</span></span><br><span class="line">/pub/software/PopLDdecay/bin/PopLDdecay -InVCF  ../data/all.vcf -SubPop  ../data/pop.SC.table -MaxDist 500 -OutStat pop.SC.stat</span><br><span class="line">/pub/software/PopLDdecay/bin/PopLDdecay -InVCF  ../data/all.vcf -SubPop ../data/pop.YZR.table -MaxDist 500 -OutStat pop.YZR.stat</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> step2.draw_singlePlot</span></span><br><span class="line">perl /pub/software/PopLDdecay/bin/Plot_OnePop.pl -inFile pop.SC.stat.gz -output pop.SC.ld</span><br><span class="line">perl /pub/software/PopLDdecay/bin/Plot_OnePop.pl -inFile pop.YZR.stat.gz -output pop.YZR.ld</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> step3.draw_multiPlotare/PopLDdecay/bin/Plot_MultiPop.pl -inList ld_stat.list -output ld_stat.multi</span></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h2 id="选择分析"><a class="header-anchor" href="#选择分析">¶</a>选择分析</h2>
<h3 id="i-ROD"><a class="header-anchor" href="#i-ROD">¶</a>i_ROD</h3>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> step1.pi</span></span><br><span class="line">vcf=../../data/all.vcf</span><br><span class="line">window=20000</span><br><span class="line">step=2000</span><br><span class="line">vcftools  --vcf $vcf --window-pi $window --window-pi-step  $step  --keep ../../data/pop.SC.table   --out ./Pi.pop1</span><br><span class="line">vcftools  --vcf $vcf --window-pi $window --window-pi-step  $step  --keep ../../data/pop.YZR.table  --out ./Pi.pop2</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> step2.ROD</span></span><br><span class="line">pi1=Pi.pop1.windowed.pi</span><br><span class="line">pi2=Pi.pop2.windowed.pi</span><br><span class="line">perl ../../script/merge2pi_ROD.pl $pi1 $pi2  &gt;Pi.ROD </span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h3 id="tajimaD"><a class="header-anchor" href="#tajimaD">¶</a>tajimaD</h3>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">vcf=../../data/all.vcf</span><br><span class="line">window=20000</span><br><span class="line"></span><br><span class="line">vcftools --vcf $vcf --TajimaD  $window  --keep  ../../data/pop.SC.table  --out TajimaD.pop1</span><br><span class="line">vcftools --vcf $vcf --TajimaD  $window  --keep  ../../data/pop.YZR.table --out   TajimaD.pop2</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h3 id="Fst"><a class="header-anchor" href="#Fst">¶</a>Fst</h3>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">vcf=../../data/all.vcf</span><br><span class="line">window=20000</span><br><span class="line">step=2000</span><br><span class="line">vcftools  --vcf $vcf --fst-window-size $window --fst-window-step $step  --weir-fst-pop  ../../data/pop.SC.table --weir-fst-pop ../../data/pop.YZR.table --out  ./Fst.pop1.pop2</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h3 id="ROD-Fst"><a class="header-anchor" href="#ROD-Fst">¶</a>ROD_Fst</h3>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">fst=../03.Fst/Fst.pop1.pop2.windowed.weir.fst</span><br><span class="line">ROD=../01.pi_ROD/Pi.ROD</span><br><span class="line">gff=../../data/genome.gff</span><br><span class="line"></span><br><span class="line">perl ../../script/top_Fst_ROD_S.pl  $fst $ROD 0.05 0.05  Fst_ROD.table Fst_ROD.stat</span><br><span class="line"></span><br><span class="line">awk &#x27;$NF==&quot;top&quot;&#x27; Fst_ROD.table  &gt; Fst_ROD.table.top</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">awk &#x27;$3==&quot;gene&quot;&#x27; $gff &gt; gene.gff</span><br><span class="line"></span><br><span class="line">bedtools  intersect -wo -F 0.1  -a  Fst_ROD.table.top -b gene.gff | awk -F &quot;\t&quot;  &#x27;&#123;print $7&quot;\t&quot;$10&quot;\t&quot;$11&quot;\t&quot;$13&quot;\t&quot;$15&#125;&#x27; | sort -u  &gt; Fst_ROD.table.top.gene</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1>GWAS</h1>
<h2 id="mpute"><a class="header-anchor" href="#mpute">¶</a>mpute</h2>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">beagle -Xmx4g  -Djava.io.tmpdir=./TMP  gt=../data/all.vcf   out=all.impute impute=true window=10 nthreads=2</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h2 id="GWAS"><a class="header-anchor" href="#GWAS">¶</a>GWAS</h2>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> step1.prepare</span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># prepare vcf</span></span></span><br><span class="line">plink --vcf ../data/all.vcf --maf 0.05 --geno 0.1  --recode12  --output-missing-genotype 0 --transpose --out snp_filter   --set-missing-var-ids @:#  --allow-extra-chr</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># prepare phenotype</span></span></span><br><span class="line"></span><br><span class="line">perl ../script/sort_pheno.pl snp_filter.tfam ../data/trait.table &gt; trait.sort.txt</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> step2.kinship</span></span><br><span class="line">/pub/software/emmax/emmax-kin-intel64 -v -d 10  -o ./pop.kinship  snp_filter</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> step3.gwas</span></span><br><span class="line">/pub/software/emmax/emmax-intel64 -v -d 10 -t snp_filter  -p trait.sort.txt -k pop.kinship   -o emmax.out 1&gt; emmax.log 2&gt;emmax.err</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> draw_manhattan</span></span><br><span class="line">paste  snp_filter.map  emmax.out.ps | awk  &#x27;BEGIN&#123;print &quot;SNP\tCHR\tBP\tP&quot;&#125;&#123;if($2==$5)&#123;print $2&quot;\t&quot;$1&quot;\t&quot;$4&quot;\t&quot;$NF&#125;&#125;&#x27;  &gt; emmax.out.ps.manht_input</span><br><span class="line">Rscript ../script/manhattan.R emmax.out.ps.manht_input  emmax.out.ps.manht_figure</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h2 id="GWAS-Q"><a class="header-anchor" href="#GWAS-Q">¶</a>GWAS-Q</h2>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> step3.gwas</span></span><br><span class="line">ln -s ../01.GWAS/snp_filter.tped</span><br><span class="line">ln -s ../01.GWAS/snp_filter.tfam</span><br><span class="line">ln -s ../01.GWAS/snp_filter.nosex</span><br><span class="line">ln -s ../01.GWAS/snp_filter.map</span><br><span class="line"></span><br><span class="line">ln -s ../01.GWAS/trait.sort.txt</span><br><span class="line">ln -s ../../02.population_genetics/03.structure/all.3.Q</span><br><span class="line"></span><br><span class="line">paste snp_filter.nosex all.3.Q |awk &#x27;&#123;print $1&quot; &quot;$1&quot; 1 &quot;$3&quot; &quot;$4&#125;&#x27;  &gt; pop.Qmatrix</span><br><span class="line"></span><br><span class="line">/pub/software/emmax/emmax-intel64 -v -d 10 -t snp_filter  -p trait.sort.txt -k pop.kinship -c pop.Qmatrix   -o emmax.out 1&gt; emmax.log 2&gt;emmax.err</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> step4.draw_manhattan</span></span><br><span class="line">paste  snp_filter.map  emmax.out.ps | awk  &#x27;BEGIN&#123;print &quot;SNP\tCHR\tBP\tP&quot;&#125;&#123;if($2==$5)&#123;print $2&quot;\t&quot;$1&quot;\t&quot;$4&quot;\t&quot;$NF&#125;&#125;&#x27;  &gt; emmax.out.ps.manht_input</span><br><span class="line">Rscript ../script/manhattan.R emmax.out.ps.manht_input  emmax.out.ps.manht_figure_Q</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h2 id="GWAS-realData"><a class="header-anchor" href="#GWAS-realData">¶</a>GWAS_realData</h2>
<figure class="highlight less"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line"># <span class="selector-tag">step1</span><span class="selector-class">.prepare</span></span><br><span class="line">## <span class="selector-tag">prepare</span> <span class="selector-tag">vcf</span></span><br><span class="line"><span class="selector-tag">plink</span> <span class="selector-tag">--vcf</span>  ../<span class="selector-tag">data_real</span>/<span class="selector-tag">Sample215</span><span class="selector-class">.M5M8H3</span><span class="selector-class">.2allel</span><span class="selector-class">.vcf</span>  <span class="selector-tag">--maf</span> <span class="selector-tag">0</span><span class="selector-class">.05</span> <span class="selector-tag">--geno</span> <span class="selector-tag">0</span><span class="selector-class">.1</span>  <span class="selector-tag">--recode12</span>  <span class="selector-tag">--output-missing-genotype</span> <span class="selector-tag">0</span> <span class="selector-tag">--transpose</span> <span class="selector-tag">--out</span> <span class="selector-tag">snp_filter</span>   <span class="selector-tag">--set-missing-var-ids</span> @:#  <span class="selector-tag">--allow-extra-chr</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">## <span class="selector-tag">prepare</span> <span class="selector-tag">phenotype</span></span><br><span class="line"></span><br><span class="line"><span class="selector-tag">perl</span> ../<span class="selector-tag">script</span>/<span class="selector-tag">sort_pheno</span><span class="selector-class">.pl</span> <span class="selector-tag">snp_filter</span><span class="selector-class">.tfam</span>  ../<span class="selector-tag">data_real</span>/<span class="selector-tag">trait</span><span class="selector-class">.C16_0</span><span class="selector-class">.table</span>  &gt; <span class="selector-tag">trait</span><span class="selector-class">.sort</span><span class="selector-class">.txt</span></span><br><span class="line"></span><br><span class="line"># <span class="selector-tag">step2</span><span class="selector-class">.kinship</span></span><br><span class="line">/<span class="selector-tag">pub</span>/<span class="selector-tag">software</span>/<span class="selector-tag">emmax</span>/<span class="selector-tag">emmax-kin-intel64</span> <span class="selector-tag">-v</span> <span class="selector-tag">-d</span> <span class="selector-tag">10</span>  <span class="selector-tag">-o</span> ./<span class="selector-tag">pop</span><span class="selector-class">.kinship</span>  <span class="selector-tag">snp_filter</span></span><br><span class="line"></span><br><span class="line"># <span class="selector-tag">step3</span><span class="selector-class">.gwas</span></span><br><span class="line">/<span class="selector-tag">pub</span>/<span class="selector-tag">software</span>/<span class="selector-tag">emmax</span>/<span class="selector-tag">emmax-intel64</span> <span class="selector-tag">-v</span> <span class="selector-tag">-d</span> <span class="selector-tag">10</span> <span class="selector-tag">-t</span> <span class="selector-tag">snp_filter</span>  <span class="selector-tag">-p</span> <span class="selector-tag">trait</span><span class="selector-class">.sort</span><span class="selector-class">.txt</span> <span class="selector-tag">-k</span> <span class="selector-tag">pop</span><span class="selector-class">.kinship</span>   <span class="selector-tag">-o</span> <span class="selector-tag">emmax</span><span class="selector-class">.out</span> <span class="selector-tag">1</span>&gt; <span class="selector-tag">emmax</span><span class="selector-class">.log</span> <span class="selector-tag">2</span>&gt;<span class="selector-tag">emmax</span><span class="selector-class">.err</span></span><br><span class="line"></span><br><span class="line"># <span class="selector-tag">step4</span><span class="selector-class">.draw_manhattan</span></span><br><span class="line"><span class="selector-tag">paste</span>  <span class="selector-tag">snp_filter</span><span class="selector-class">.map</span>  <span class="selector-tag">emmax</span><span class="selector-class">.out</span><span class="selector-class">.ps</span> | <span class="selector-tag">awk</span>  &#x27;<span class="selector-tag">BEGIN</span>&#123;<span class="selector-tag">print</span> &quot;<span class="selector-tag">SNP</span>\<span class="selector-tag">tCHR</span>\<span class="selector-tag">tBP</span>\<span class="selector-tag">tP</span>&quot;&#125;&#123;<span class="selector-tag">if</span>($<span class="number">2</span>==$<span class="number">5</span>)&#123;<span class="selector-tag">print</span> $<span class="selector-tag">2</span>&quot;\<span class="selector-tag">t</span>&quot;$<span class="selector-tag">1</span>&quot;\<span class="selector-tag">t</span>&quot;$<span class="selector-tag">4</span>&quot;\<span class="selector-tag">t</span>&quot;$<span class="selector-tag">NF</span>&#125;&#125;&#x27;  &gt; <span class="selector-tag">emmax</span><span class="selector-class">.out</span><span class="selector-class">.ps</span><span class="selector-class">.manht_input</span></span><br><span class="line"><span class="selector-tag">Rscript</span> ../<span class="selector-tag">script</span>/<span class="selector-tag">manhattan</span><span class="selector-class">.R</span> <span class="selector-tag">emmax</span><span class="selector-class">.out</span><span class="selector-class">.ps</span><span class="selector-class">.manht_input</span>  <span class="selector-tag">emmax</span><span class="selector-class">.out</span><span class="selector-class">.ps</span><span class="selector-class">.manht_figure_readData</span></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1>QTL-seq</h1>
<h2 id="change-VCF2Table"><a class="header-anchor" href="#change-VCF2Table">¶</a>change_VCF2Table</h2>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">ref=</span><br><span class="line">vcf=</span><br><span class="line">outfile=</span><br><span class="line">gatk  VariantsToTable -R  $ref -V $vcf -F CHROM -F POS -F REF -F ALT -GF AD -GF DP -GF GQ -GF PL -O  $outfile</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h2 id="QTLSeq-Analysis"><a class="header-anchor" href="#QTLSeq-Analysis">¶</a>QTLSeq_Analysis</h2>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br></pre></td><td class="code"><pre><span class="line">library(QTLseqr)</span><br><span class="line">library(<span class="string">&quot;ggplot2&quot;</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">HighBulk &lt;- <span class="string">&quot;SRR834931&quot;</span> <span class="comment">## 高值混池名称</span></span><br><span class="line">LowBulk &lt;- <span class="string">&quot;SRR834927&quot;</span>  <span class="comment">## 低值混池名称</span></span><br><span class="line">Chroms &lt;- paste0(<span class="built_in">rep</span>(<span class="string">&quot;Chr&quot;</span>, <span class="number">12</span>), <span class="number">1</span>:<span class="number">12</span>) <span class="comment">## 染色体列表</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## SNP数据读取</span></span><br><span class="line">df &lt;- importFromGATK(file = <span class="string">&quot;../data/Yang_et_al_2013.table&quot;</span> ,</span><br><span class="line">    highBulk = HighBulk, <span class="comment"># 指定高值混池</span></span><br><span class="line">    lowBulk = LowBulk, <span class="comment"># 指定低至混池</span></span><br><span class="line">    chromList = Chroms <span class="comment"># 指定分析用染色体列表</span></span><br><span class="line">    ) </span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## 绘制深度分布图 </span></span><br><span class="line"></span><br><span class="line">p1 &lt;- ggplot(data = df) +</span><br><span class="line">  geom_histogram(aes(x = DP.HIGH + DP.LOW)) +</span><br><span class="line">  xlim(<span class="number">0</span>,<span class="number">800</span>)</span><br><span class="line">  </span><br><span class="line">pdf(file=<span class="string">&quot;SNP_depth.pdf&quot;</span>)</span><br><span class="line">p1</span><br><span class="line">dev.off()</span><br><span class="line"></span><br><span class="line"><span class="comment">## 绘制ref等位基因频率分布图</span></span><br><span class="line">p2 &lt;- ggplot(data = df) +</span><br><span class="line">  geom_histogram(aes(x = REF_FRQ))</span><br><span class="line">  </span><br><span class="line">pdf(file=<span class="string">&quot;ref_allele_frequency.pdf&quot;</span>)</span><br><span class="line">p2</span><br><span class="line">dev.off()</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## 绘制高值混池SNP-index分布</span></span><br><span class="line">p3 &lt;- ggplot(data = df) +</span><br><span class="line">  geom_histogram(aes(x = SNPindex.HIGH))</span><br><span class="line">pdf(file=<span class="string">&quot;SNPindex.HIGH.dis.pdf&quot;</span>)</span><br><span class="line">p3</span><br><span class="line">dev.off()</span><br><span class="line"></span><br><span class="line"><span class="comment">## 绘制低值混池SNP-index分布</span></span><br><span class="line">p4 &lt;- ggplot(data = df) +</span><br><span class="line">  geom_histogram(aes(x = SNPindex.LOW))</span><br><span class="line"></span><br><span class="line">pdf(file=<span class="string">&quot;SNPindex.LOW.dis.pdf&quot;</span>)</span><br><span class="line">p4</span><br><span class="line">dev.off()</span><br><span class="line"></span><br><span class="line"><span class="comment">## SNP过滤</span></span><br><span class="line"></span><br><span class="line">df_filt &lt;- filterSNPs(</span><br><span class="line">    SNPset = df,</span><br><span class="line">    refAlleleFreq = <span class="number">0.20</span>, <span class="comment">## ref allele频率过滤，0.2表示0.2~0.8之间</span></span><br><span class="line">    minTotalDepth = <span class="number">100</span>, <span class="comment">## 最小深度过滤</span></span><br><span class="line">    maxTotalDepth = <span class="number">400</span>, <span class="comment">## 最大深度过滤</span></span><br><span class="line">    minSampleDepth = <span class="number">40</span>, <span class="comment">## 单个样品最小深度</span></span><br><span class="line">    minGQ = <span class="number">99</span>, <span class="comment">## genotype quality 过滤</span></span><br><span class="line">    verbose = <span class="literal">TRUE</span>  <span class="comment">## 输出日志</span></span><br><span class="line">  )</span><br><span class="line"></span><br><span class="line"><span class="comment">## 进行deltaSNPindex 计算</span></span><br><span class="line">df_filt &lt;- runQTLseqAnalysis(df_filt,</span><br><span class="line">    windowSize = <span class="number">1e6</span>, <span class="comment">## 窗口大小</span></span><br><span class="line">    popStruc = <span class="string">&quot;F2&quot;</span>,  <span class="comment">## 群体类型，F2 或者RIL</span></span><br><span class="line">    bulkSize = <span class="built_in">c</span>(<span class="number">385</span>, <span class="number">430</span>), <span class="comment">## 混样个数，第一个高值组</span></span><br><span class="line">    replications = <span class="number">10000</span>, <span class="comment">## bootstrap次数</span></span><br><span class="line">    intervals = <span class="built_in">c</span>(<span class="number">95</span>, <span class="number">99</span>) <span class="comment">## 置信区间</span></span><br><span class="line">)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## SNP 沿染色体分布图</span></span><br><span class="line">p5 &lt;- plotQTLStats(SNPset = df_filt, var = <span class="string">&quot;nSNPs&quot;</span>)</span><br><span class="line">pdf(file=<span class="string">&quot;SNP_filter.window.pdf&quot;</span>, width = <span class="number">20</span>, height = <span class="number">4</span>)</span><br><span class="line">p5</span><br><span class="line">dev.off()</span><br><span class="line"></span><br><span class="line"><span class="comment">## deltaSNPindex沿染色体分布</span></span><br><span class="line">p6 &lt;- plotQTLStats(SNPset = df_filt, var = <span class="string">&quot;deltaSNP&quot;</span>, plotIntervals = <span class="literal">TRUE</span>)</span><br><span class="line">pdf(file=<span class="string">&quot;deltaSNPindex.pdf&quot;</span>, width = <span class="number">20</span>, height = <span class="number">4</span>)</span><br><span class="line">p6</span><br><span class="line">dev.off()</span><br><span class="line"></span><br><span class="line"><span class="comment">## 提取显著性区域</span></span><br><span class="line">QTL &lt;- getSigRegions(SNPset = df_filt, </span><br><span class="line">    method = <span class="string">&quot;QTLseq&quot;</span>, interval = <span class="number">95</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">## 输出到文件</span></span><br><span class="line">write.table(QTL[[<span class="number">1</span>]],</span><br><span class="line">    <span class="string">&quot;QTLseq_result.SigRegions.table&quot;</span>,</span><br><span class="line">    sep=<span class="string">&quot;\t&quot;</span>,<span class="built_in">quote</span>=<span class="built_in">F</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">## 提取QTLseq结果</span></span><br><span class="line">results99 &lt;- getQTLTable(</span><br><span class="line">  SNPset = df_filt,  </span><br><span class="line">  method = <span class="string">&quot;QTLseq&quot;</span>, </span><br><span class="line">  interval = <span class="number">99</span>,  <span class="comment">## 结果阈值</span></span><br><span class="line">  export = <span class="literal">FALSE</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">## 输出到文件</span></span><br><span class="line">write.table(results99,</span><br><span class="line">    file=<span class="string">&quot;QTLseq_result_deltaSNPindex_CI99.table&quot;</span> ,</span><br><span class="line">    sep=<span class="string">&quot;\t&quot;</span>, <span class="built_in">quote</span> = <span class="built_in">F</span>)</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<hr>
<blockquote>
<p>💌lixiang117423@gmail.com</p>
<p>💌lixiang117423@foxmail.com</p>
</blockquote>
</article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">Author: </span><span class="post-copyright-info"><a href="mailto:undefined">小蓝哥</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">Link: </span><span class="post-copyright-info"><a href="http://example.com/2021/11/23/%E9%87%8D%E6%B5%8B%E5%BA%8F%E5%88%86%E6%9E%90%E8%84%9A%E6%9C%AC/">http://example.com/2021/11/23/%E9%87%8D%E6%B5%8B%E5%BA%8F%E5%88%86%E6%9E%90%E8%84%9A%E6%9C%AC/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">Copyright Notice: </span><span class="post-copyright-info">All articles in this blog are licensed under <a target="_blank" rel="noopener" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a> unless stating additionally.</span></div></div><div class="tag_share"><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/Linux/">Linux</a><a class="post-meta__tags" href="/tags/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6/">生物信息学</a></div><div class="post_share"><div class="social-share" data-image="/img/cover/popgene.jpg" data-sites="facebook,twitter,wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/social-share.js/dist/js/social-share.min.js" defer></script></div></div><div class="post-reward"><div class="reward-button button--animated"><i class="fas fa-qrcode"></i> Donate</div><div class="reward-main"><ul class="reward-all"><li class="reward-item"><a href="/img/wechat.jpg" target="_blank"><img class="post-qr-code-img" src="/img/wechat.jpg" alt="微信"/></a><div class="post-qr-code-desc">微信</div></li><li class="reward-item"><a href="/img/alipay.jpg" target="_blank"><img class="post-qr-code-img" src="/img/alipay.jpg" alt="支付宝"/></a><div class="post-qr-code-desc">支付宝</div></li></ul></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/2021/11/23/NCBI-blast-%E6%9C%AC%E5%9C%B0%E6%AF%94%E5%AF%B9/"><img class="prev-cover" src="/img/cover/blast.jpg" onerror="onerror=null;src='/img/404.jpg'" alt="cover of previous post"><div class="pagination-info"><div class="label">Previous Post</div><div class="prev_info">NCBI-blast-本地比对</div></div></a></div><div class="next-post pull-right"><a href="/2021/07/06/GENIE3%E6%9E%84%E5%BB%BA%E5%9F%BA%E5%9B%A0%E8%B0%83%E6%8E%A7%E7%BD%91%E7%BB%9C/"><img class="next-cover" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/img/default.jpg" onerror="onerror=null;src='/img/404.jpg'" alt="cover of next post"><div class="pagination-info"><div class="label">Next Post</div><div class="next_info">GENIE3构建基因调控网络</div></div></a></div></nav><div class="relatedPosts"><div class="headline"><i class="fas fa-thumbs-up fa-fw"></i><span>Related Articles</span></div><div class="relatedPosts-list"><div><a href="/2021/04/05/Linux-Tips/" title="Linux Tips"><img class="cover" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/img/default.jpg" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2021-04-05</div><div class="title">Linux Tips</div></div></a></div><div><a href="/2021/03/01/WGS%E5%AD%A6%E4%B9%A0%E7%AC%94%E8%AE%B0/" title="WGS学习笔记"><img class="cover" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/img/default.jpg" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2021-03-01</div><div class="title">WGS学习笔记</div></div></a></div><div><a href="/2021/06/01/iTAK%E7%AE%80%E6%98%8E%E6%95%99%E7%A8%8B/" title="iTAK简明教程"><img class="cover" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/img/default.jpg" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2021-06-01</div><div class="title">iTAK简明教程</div></div></a></div><div><a href="/2020/01/09/iTOL%E4%BF%AE%E9%A5%B0%E8%BF%9B%E5%8C%96%E6%A0%91/" title="iTOL修饰进化树"><img class="cover" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/img/default.jpg" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2020-01-09</div><div class="title">iTOL修饰进化树</div></div></a></div><div><a href="/2021/07/03/%C2%96Win10%E5%AD%90%E7%B3%BB%E7%BB%9F%E5%AE%89%E8%A3%85%E8%BF%90%E8%A1%8CQIIME2/" title="Win10子系统安装运行QIIME2"><img class="cover" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/img/default.jpg" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2021-07-03</div><div class="title">Win10子系统安装运行QIIME2</div></div></a></div><div><a href="/2020/01/07/%E9%80%9A%E8%BF%87Linux%E5%91%BD%E4%BB%A4%E8%A1%8C%E4%BD%BF%E7%94%A8Aspera%E5%85%A8%E9%80%9F%E4%B8%8A%E4%BC%A0%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE%E5%88%B0NCBI%E6%95%B0%E6%8D%AE%E5%BA%93/" title="通过Linux命令行使用Aspera全速上传测序数据到NCBI数据库"><img class="cover" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/img/default.jpg" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2020-01-07</div><div class="title">通过Linux命令行使用Aspera全速上传测序数据到NCBI数据库</div></div></a></div></div></div><hr/><div id="post-comment"><div class="comment-head"><div class="comment-headline"><i class="fas fa-comments fa-fw"></i><span> Comment</span></div></div><div class="comment-wrap"><div><div class="vcomment" id="vcomment"></div></div></div></div></div><div class="aside-content" id="aside-content"><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>Catalog</span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link"><span class="toc-number">1.</span> <span class="toc-text">软件安装</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#lumpy-sv"><span class="toc-number">1.1.</span> <span class="toc-text">lumpy-sv</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#cnvnator"><span class="toc-number">1.2.</span> <span class="toc-text">cnvnator</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link"><span class="toc-number">2.</span> <span class="toc-text">变异检测</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%9E%84%E5%BB%BA%E5%9F%BA%E5%9B%A0%E7%BB%84index"><span class="toc-number">2.1.</span> <span class="toc-text">构建基因组index</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%AF%94%E5%AF%B9%E6%8E%92%E5%BA%8F%E5%8E%BB%E9%87%8D"><span class="toc-number">2.2.</span> <span class="toc-text">比对排序去重</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#SNP%E3%80%81Indel%E6%A3%80%E6%B5%8B"><span class="toc-number">2.3.</span> <span class="toc-text">SNP、Indel检测</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#SV%E6%A3%80%E6%B5%8B"><span class="toc-number">2.4.</span> <span class="toc-text">SV检测</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#CNV%E6%A3%80%E6%B5%8B"><span class="toc-number">2.5.</span> <span class="toc-text">CNV检测</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#SNP-INDEL%E6%B3%A8%E9%87%8A"><span class="toc-number">2.6.</span> <span class="toc-text">SNP INDEL注释</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#CNV%E5%92%8CSV%E6%B3%A8%E9%87%8A"><span class="toc-number">2.7.</span> <span class="toc-text">CNV和SV注释</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link"><span class="toc-number">3.</span> <span class="toc-text">群体结构</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E8%BF%87%E6%BB%A4"><span class="toc-number">3.1.</span> <span class="toc-text">过滤</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%BB%BA%E6%A0%91"><span class="toc-number">3.2.</span> <span class="toc-text">建树</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#NJ"><span class="toc-number">3.2.1.</span> <span class="toc-text">NJ</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#tree-snphylo"><span class="toc-number">3.2.2.</span> <span class="toc-text">tree_snphylo</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#PCA"><span class="toc-number">3.3.</span> <span class="toc-text">PCA</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E7%BE%A4%E4%BD%93%E7%BB%93%E6%9E%84"><span class="toc-number">3.4.</span> <span class="toc-text">群体结构</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E9%81%97%E4%BC%A0%E8%A1%B0%E5%87%8F"><span class="toc-number">3.5.</span> <span class="toc-text">遗传衰减</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E9%80%89%E6%8B%A9%E5%88%86%E6%9E%90"><span class="toc-number">3.6.</span> <span class="toc-text">选择分析</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#i-ROD"><span class="toc-number">3.6.1.</span> <span class="toc-text">i_ROD</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#tajimaD"><span class="toc-number">3.6.2.</span> <span class="toc-text">tajimaD</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Fst"><span class="toc-number">3.6.3.</span> <span class="toc-text">Fst</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#ROD-Fst"><span class="toc-number">3.6.4.</span> <span class="toc-text">ROD_Fst</span></a></li></ol></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link"><span class="toc-number">4.</span> <span class="toc-text">GWAS</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#mpute"><span class="toc-number">4.1.</span> <span class="toc-text">mpute</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#GWAS"><span class="toc-number">4.2.</span> <span class="toc-text">GWAS</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#GWAS-Q"><span class="toc-number">4.3.</span> <span class="toc-text">GWAS-Q</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#GWAS-realData"><span class="toc-number">4.4.</span> <span class="toc-text">GWAS_realData</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link"><span class="toc-number">5.</span> <span class="toc-text">QTL-seq</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#change-VCF2Table"><span class="toc-number">5.1.</span> <span class="toc-text">change_VCF2Table</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#QTLSeq-Analysis"><span class="toc-number">5.2.</span> <span class="toc-text">QTLSeq_Analysis</span></a></li></ol></li></ol></div></div></div></div></main><footer id="footer" style="background-image: url('/img/cover/popgene.jpg')"><div id="footer-wrap"><div class="copyright">&copy;2021 By 小蓝哥</div><div class="framework-info"><span>Framework </span><a target="_blank" rel="noopener" href="https://hexo.io">Hexo</a><span class="footer-separator">|</span><span>Theme </span><a target="_blank" rel="noopener" href="https://github.com/jerryc127/hexo-theme-butterfly">Butterfly</a></div></div></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="Read Mode"><i class="fas fa-book-open"></i></button><button id="font-plus" type="button" title="Increase font size"><i class="fas fa-plus"></i></button><button id="font-minus" type="button" title="Decrease font size"><i class="fas fa-minus"></i></button><button id="translateLink" type="button" title="Toggle Between Traditional Chinese And Simplified Chinese">簡</button><button id="darkmode" type="button" title="Toggle Between Light And Dark Mode"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="Toggle between single-column and double-column"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="Setting"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="Table Of Contents"><i class="fas fa-list-ul"></i></button><a id="to_comment" href="#post-comment" title="Scroll To Comments"><i class="fas fa-comments"></i></a><button id="go-up" type="button" title="Back To Top"><i class="fas fa-arrow-up"></i></button></div></div><div id="local-search"><div class="search-dialog"><div class="search-dialog__title" id="local-search-title">Local search</div><div id="local-input-panel"><div id="local-search-input"><div class="local-search-box"><input class="local-search-box--input" placeholder="Search for Posts" type="text"/></div></div></div><hr/><div id="local-search-results"></div><span class="search-close-button"><i class="fas fa-times"></i></span></div><div id="search-mask"></div></div><div><script src="/js/utils.js"></script><script src="/js/main.js"></script><script src="/js/tw_cn.js"></script><script src="https://cdn.jsdelivr.net/npm/node-snackbar/dist/snackbar.min.js"></script><script src="/js/search/local-search.js"></script><script>var preloader = {
  endLoading: () => {
    document.body.style.overflow = 'auto';
    document.getElementById('loading-box').classList.add("loaded")
  },
  initLoading: () => {
    document.body.style.overflow = '';
    document.getElementById('loading-box').classList.remove("loaded")

  }
}
window.addEventListener('load',preloader.endLoading())</script><div class="js-pjax"><script>function loadValine () {
  function initValine () {
    const valine = new Valine(Object.assign({
      el: '#vcomment',
      appId: '',
      appKey: '',
      placeholder: 'Please leave your footprints',
      avatar: 'monsterid',
      meta: 'nick,mail,link'.split(','),
      pageSize: '10',
      lang: 'en',
      recordIP: false,
      serverURLs: '',
      emojiCDN: '',
      emojiMaps: "",
      enableQQ: false,
      path: window.location.pathname,
      requiredFields: ["nick,mail"],
      visitor: false
    }, null))
  }

  if (typeof Valine === 'function') initValine() 
  else getScript('https://cdn.jsdelivr.net/npm/valine/dist/Valine.min.js').then(initValine)
}

if ('Valine' === 'Valine' || !true) {
  if (true) btf.loadComment(document.getElementById('vcomment'),loadValine)
  else setTimeout(loadValine, 0)
} else {
  function loadOtherComment () {
    loadValine()
  }
}</script></div><canvas class="fireworks" mobile="false"></canvas><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/dist/fireworks.min.js"></script><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/dist/activate-power-mode.min.js"></script><script>POWERMODE.colorful = true;
POWERMODE.shake = true;
POWERMODE.mobile = false;
document.body.addEventListener('input', POWERMODE);
</script><script async data-pjax src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div></body></html>