---
title: Rè¯­è¨€å¸¸ç”¨Tips
tags: Rè¯­è¨€
categories: Rè¯­è¨€
top: true
abbrlink: '15541358'
date: 2021-01-18 17:17:33
---

ç»å¸¸è¦ç”¨åˆ°`R`ï¼Œæœ‰äº›å°æŠ€å·§æ¯æ¬¡éƒ½è¦å»æŸ¥ï¼Œæ¯”è¾ƒéº»çƒ¦ï¼Œå¹²è„†è®°å½•ä¸€ä¸‹ã€‚

<!-- more -->

# è½¯ä»¶å®‰è£…

1. å®‰è£…`Rtools`ï¼š

   - å®‰è£… Rtools4.0ï¼Œå®‰è£…åŒ…ï¼š[https://cran.r-project.org/bin/windows/Rtools/](https://link.zhihu.com/?target=https%3A//cran.r-project.org/bin/windows/Rtools/)

   - é…ç½®ç¯å¢ƒ

     - åœ¨ RStudio é‡Œé¢è¿è¡Œä»¥ä¸‹è„šæœ¬ï¼š

       ```R
       writeLines('PATH="${RTOOLS40_HOME}\\usr\\bin;${PATH}"', con = "~/.Renviron")
       ```

     - é‡æ–°å¯åŠ¨ RStudioï¼Œç„¶åè¿è¡Œä»¥ä¸‹ä»£ç ï¼š

       ```R
       Sys.which("make")
       ```

       ä¼šå¾—åˆ°ç»“æœï¼š"C:\\rtools40\\usr\\bin\\make.exe"ï¼ˆä¹Ÿå°±æ˜¯ make.exe çš„è·¯å¾„ï¼‰

     - å°è¯•å®‰è£…ä¸€ä¸ªåŒ…

       ```R
       install.packages("jsonlite", type = "source")
       ```

2. `RStudio`ä¸­æ–‡ä¹±ç ï¼š
   èœå•æ ä¸­çš„ `file->reopen with encoding->utf-8`

3. `Rshiny`ä¸Šä¼ æ–‡ä»¶å¤§å°é™åˆ¶ï¼šåœ¨`server.R`æ–‡ä»¶é¡¶éƒ¨åŠ ä¸Šä¸‹é¢è¿™è¡Œä»£ç ï¼š

   ```R
   options(shiny.maxRequestSize=1000*1024^2)
   ```
   
4. æ›´æ–°æ‰€æœ‰RåŒ…ï¼š

   ```R
   #å®‰è£…åŒ…
   install.packages("rvcheck")
   #åŠ è½½åŒ…
   library(rvcheck)
   #æ£€æŸ¥Ræ˜¯å¦æœ‰æ›´æ–°
   rvcheck::check_r()
   # æ›´æ–°æ‰€æœ‰RåŒ…
   rvcheck::update_all(check_R = FALSE,which = c("CRAN","BioC","github")
   ```
   
5. æ›´æ–°Rç‰ˆæœ¬ï¼š

   ```R
   library(installr)
   updateR()
   ```

   

# `ggplot2`

1. `ggplot2`é™åˆ¶Yè½´èŒƒå›´ï¼š

   ```R
   coord_cartesian(ylim = c(5, 22))
   ```

2. `ggplot2`è¾“å‡ºä¸­æ–‡ï¼š

   ```R
    theme_bw(base_family = "STKaiti")
   ```

3. `ggplot2`åæ ‡è½´æ–‡å­—è§’åº¦ï¼š

   ```R
   axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)
   ```

4. `ggplot2`å¯¼å‡ºæˆ`PPT`ï¼š

   ```R
   export::graph2ppt(p2, width = 6, height = 5,
                     margins = c(top = 0.5, right = 0.5, 
                                     bottom = 0.5, left= 0.5), 
                     center = TRUE,
                     file = '../results/PPT/ç¨»ç˜Ÿç—…å‘ç—…æƒ…å†µï¼ˆ2020ï¼‰.ppt')
   ```

5. `ggplot2`æ¸å˜è‰²å¡«å……ï¼š

   ```R
   scale_fill_gradient2(low = "#000080", mid = "white", high = "#B22222")
   ```
   
6. å›¾ä¾‹åç§°ï¼š

   ```R
   labs(fill = 'Cor')
   ```
   
7. åæ ‡è½´åˆ»åº¦é•¿åº¦ï¼š

   ```R
   theme(axis.ticks.length.y = unit(0,'mm'))
   ```

8. é¢ å€’Xè½´å’ŒYè½´ï¼š

   ```R
   coord_flip()
   ```

   

# çƒ­å›¾ç›¸å…³

1. `ComplexHeatmap`æ³¨é‡Šçš„é¢œè‰²è®¾ç½®ï¼š

   ```R
   ann.right = HeatmapAnnotation(which = 'row',
                                 gap = unit(1.5,'mm'),
                                 `Class I` = res.cor.3$`Class I`,
                                 annotation_name_side = 'top',
                                 annotation_name_rot = 90,
                                 annotation_name_offset = unit(2,'mm'),
                                 col = list(`Class I` = c('Lipids' = '#DC143C',
                                                         'Phenolic acids' = '#00FF00',
                                                         'Alkaloids' = '#483D8B',
                                                         'Terpenoids' = '#808000',
                                                         'Organic acids' = '#FF8C00',
                                                         'Others' = '#B22222',
                                                         'Nucleotides' = '#000000',
                                                         'Flavonoids' = '#006400',
                                                         'Amino acids' = '#0000CD')))
   ```

   

2. è°ƒç”¨`pheatmap`åŒ…ä¸­çš„å‡½æ•°è§£å†³`ComplexHeatmap::Heatmap`æ•°æ®æ ‡å‡†åŒ–é—®é¢˜ï¼š

   ```R
   df.heatmap[,1:6] %>% as.matrix() %>%
     pheatmap:::scale_rows() %>%
     ComplexHeatmap::Heatmap(name = 'ç›¸ å¯¹\nè¡¨è¾¾é‡',
                             col = c("navy", "white", "firebrick3"),
                             show_row_names = FALSE)
   ```
   
3. çƒ­å›¾æ ‡æ³¨æ˜¾è‘—æ€§ï¼š

   ```R
   p.cor = quickcor(design, mes, cor.test = TRUE) +
     geom_colour() +
     geom_mark(r = NA,sig.thres = 0.05, 
               size = 5, color = 'black',
               nudge_x = 0, nudge_y = 0) +
     scale_fill_gradient2(low = "navy", 
                          mid = "white", 
                          high = "firebrick3")+
     labs(fill = 'Cor')
   ```

4. `ggcor`ç»˜åˆ¶ä¸¤ä¸ªçŸ©é˜µçš„çƒ­å›¾ï¼š

   ```R
   quickcor(df.dems.1, df.degs.1.down, cor.test = TRUE, height = 10) +
       geom_colour() +
       scale_fill_gradient2(low = "navy", 
                            mid = "white", 
                            high = "firebrick3")+
       labs(fill = 'Cor')+
       remove_x_axis() +
       theme(axis.ticks.length.y = unit(0,'mm'),
             axis.text.y = element_text(size = 1))
   ```

   

# æ•°æ®å¤„ç†

1. `dcast()`å‡½æ•°ç”¨æ³•ï¼š

   ```R
   reshape2::dcast(res.cor, gene ~ meta) # geneæ˜¯è¡Œï¼Œmetaæ˜¯åˆ—
   # acast()å‡½æ•°çš„è¯ç›´æ¥å°±ç”Ÿæˆrownames
   # geneåœ¨ç¬¬ä¸€åˆ—ï¼Œmetaåœ¨ç¬¬äºŒåˆ—ï¼Œè¿™æ ·æ‰èƒ½æˆåŠŸï¼Œæˆ‘ä¹Ÿä¸çŸ¥é“æ˜¯ä¸ºå•¥ï¼
   ```

2. æ–¹å·®åˆ†æå¤šé‡æ¯”è¾ƒï¼š

   ```R
   tuk.15 = glht(fit.15, linfct = mcp(donor = 'Tukey'))
   sig.15 = cld(tuk.15, level = 0.95, decreasing = TRUE)[["mcletters"]][["Letters"]] %>%
           as.data.frame()
   ```

>ğŸ’Œlixiang117423@gmail.com
>
>ğŸ’Œlixiang117423@foxmail.com

